### ISSUE TITLE
Configure poll intervall

### ISSUE THREAD

**Issue by fredo** [beamer-bridge/beamer#1368]
Poll interval is configured to 5 seconds hard coded. We may want to give agents the option to choose the interval.

However, we should be careful with the right moment when doing this as it exposes a centralization risk. Faster poll intervals will likely get more transfers

**Comment by istankovic** [comment]
> However, we should be careful with the right moment when doing this as it exposes a centralization risk. Faster poll intervals will likely get more transfers

Note that it is currently easy to change the poll interval for an agent:
```
git clone <repo>
# modify the poll period
make container-image
```
and then just use the resulting image.

Being able to set the period via e.g. a config file would be somewhat easier & faster, but I don't think it would increase the centralization risk in a real way (people who understand enough to want to set the option are presumably also capable of doing the above sequence of commands).

**Comment by fredo** [comment]
Don't underestimate humans laziness and sticking to the default ;)

I thought a bit more about it and actually I thought about an alternative argument that promotes adding this feature sooner rather than later.

Of course the centralization risk lies in the underlying protocol design and not in the open source agent software which anybody can change (or even write its own, as we already got feedback for this).
But before we can tackle solving this issue fundamentally, we have to deal with the current situation.

First I thought that even though it is not too hard to change it, we should not actively promote the centralization risk, as by adding a config parameter, it is as easy as changing one number, plus the agent has to touch the config anyway. As a matter of fact that lowers the entry barrier for centralization.

On the other hand now, I came to the conclusion that BY offering to configure this value, we give everybody the chance to lower the poll interval, which could have a "decentralization" effect or at least does actually prevent centralization. It is so easy that any agent can do it.

If we leave it as it is, people who take the effort, will be superior and the likelihood of only one agent doing it is higher thus centralization is more likely than allowing to configure this value easily. 
What do you think?

**Comment by istankovic** [comment]
> On the other hand now, I came to the conclusion that BY offering to configure this value, we give everybody the chance to lower the poll interval, which could have a "decentralization" effect or at least does actually prevent centralization. It is so easy that any agent can do it.

I love this -- "let's make a market" thing. :-)
It's so similar to flashbots and other such constructs in the space.

> If we leave it as it is, people who take the effort, will be superior and the likelihood of only one agent doing it is higher thus centralization is more likely than allowing to configure this value easily.
What do you think?

I think we should definitely add the mechanism, we probably should already have.

You definitely have a point regarding the ease of configuring this. It's just that I feel that whoever is motivated to do it, it does not really matter to them since even what we have now is not hard at all (it would be a different thing if it were much harder, naturally). What we're planning to do here will lower the entry barrier, but the real effect remains to be seen.

In any case, I definitely would do this. :+1:

**Comment by fredo** [comment]
this is already done
